### PR TITLE
Add array_overlap

### PR DESCRIPTION
**Pull request by Pslydhh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

return boolean to indicate that if two arrays have intersection.

### CODE REVIEW

File: be/src/exprs/vectorized/array_functions.tpp [hunk 1]
```diff
@@ -105,6 +105,102 @@ private:
...
+        if (is_nullable) {
+            return NullableColumn::create(result_column, null_result);
+        }
+
+        return result_column;
+    }
+
+    template <typename HashSet>
+    static void _array_overlap_item(const std::vector<ArrayColumn*>& columns, size_t index, uint8_t* data) {
+        HashSet hash_set;
```
Comment (kangkaisen) [inline be/src/exprs/vectorized/array_functions.tpp:166]: The hash_set could be reused.

File: be/src/exprs/vectorized/array_functions.tpp [hunk 2]
```diff
@@ -105,6 +105,102 @@ private:
...
+            return _array_overlap<phmap::flat_hash_set<CppType, SliceHash>>(columns);
+        } else {
+            assert(false);
+        }
+    }
+
+private:
+    template <typename HashSet>
+    static ColumnPtr _array_overlap(const Columns& columns) {
+        DCHECK_EQ(columns.size(), 2);
```
Comment (kangkaisen) [inline be/src/exprs/vectorized/array_functions.tpp:128]: You should check the arg number in FE

File: be/src/exprs/vectorized/array_functions.h [hunk 3]
```diff
@@ -40,6 +40,27 @@ class ArrayFunctions {
 
 #undef DEFINE_ARRAY_DISTINCT_FN
 
+#define DEFINE_ARRAY_OVERLAP_FN(NAME, PT)                                                     \
+    static ColumnPtr array_overlap_##NAME(FunctionContext* context, const Columns& columns) { \
+        return ArrayOverlap<PT>::process(context, columns);                                   \
+    }
+
+    DEFINE_ARRAY_OVERLAP_FN(boolean, PrimitiveType::TYPE_BOOLEAN);
```
Comment (mofeiatwork) [inline be/src/exprs/vectorized/array_functions.h:48]: the type enumeration code could be replaced with `olap_type_infra.h`

Comment (Pslydhh) [reply be/src/exprs/vectorized/array_functions.h:48]: > the type enumeration code could be replaced with `olap_type_infra.h`

Micro in olap_type_infra.h is dispatch with FieldType, codes here is dispatch with PrimitiveType. could I add this micro to olap_type_infra.h?

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**